pallets/did/src/migrations/v1.rs

```diff
@@ -0,0 +1,189 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate::*;
+
+use sp_runtime::traits::Zero;
+
+#[cfg(feature = "try-runtime")]
+pub(crate) fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+	ensure!(
+		StorageVersion::<T>::get() == DidStorageVersion::V1,
+		"Current deployed version is not v1."
+	);
+
+	log::info!("Version storage migrating from v1 to v2");
+	Ok(())
+}
+
+pub(crate) fn migrate<T: Config>() -> Weight {
+	log::info!("v1 -> v2 DID storage migrator started!");
+	let mut total_weight = Weight::zero();
+
+	Did::<T>::translate_values(|old_details| Some(old_to_new_did_details(old_details)));
+
+	StorageVersion::<T>::set(DidStorageVersion::V2);
+	// Adds a write from StorageVersion::set() weight.
+	total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+	log::debug!("Total weight consumed: {}", total_weight);
+	log::info!("v1 -> v2 DID storage migrator finished!");
+
+	total_weight
+}
+
+fn old_to_new_did_details<T: Config>(old: deprecated::v1::DidDetails<T>) -> DidDetails<T> {
```

Comment:
I think we might not even need a migration since we just remove one field and initiate a new one with default `None`. But better to be safe.

Reply:
we need a migration here in case there are entries with a ServiceEndpointUrl. The new code would try to parse the URL into the new Struct.